docs/way-of-working/guides/testing/testing.md

```diff
@@ -1 +1,132 @@
-## Testing overview
\ No newline at end of file
+---
+title: Way of Testing
+sidebar_position: 10
+---
+
+:::danger[Draft]
+
+This document is still in draft. You can discuss it as part of this PR: https://github.com/dhis2/wow-frontend/pull/7.
+
+The scope of this document is automated tests _in code_; unit, integration, end to end, contract etc... in the context of the _frontend_ for platform team. It does not tackle other important aspects such as manual testing and its processes, or backend and database tests.
+:::
+
+### Why automated tests?
+
+Why we write tests? Good tests can...
+
+1. Verify the code is working correctly...
+2. Prevent future regressions
+3. Document code behaviour
+4. Provides design guidance
+5. Support refactoring
+
+> Read the full article here:  https://madeintandem.com/blog/five-factor-testing/
+
+
+:::warning[Discussion]
+
+- Rank these reasons in terms of importance in our context according to you
+- Any other reasons we should write tests?
+
+:::
+
+
+### The ideal scenario
+
+(Mozafar): we have a **high coverage** of tests, _integration-like_ with the most crucial scenarios tested in a thin layer of _end to end_ tests. Regressions and breaking changes in the API are caught early by tests, and refactoring made easy.
+
+:::warning[Discussion]
+
+- How does the ideal scenario look for you? some other possible prompts: TDD, cucumber, contract testing, end to end, pair programming... 
+
+:::
+
+:::warning[Discussion]
+
+- What should be a high coverage in our context?
```

Comment:
Having 0% coverage is a very obvious sign that we need more tests.

But having a close-to-100% coverage comes with its own downsides:
* We'd be testing things that don't need tests
* Changing code can become very cumbersome because that means a lot of tests have to be addressed

So I'm not sure what a good number would be. But if we want to do mostly integration tests - and in an ideal case we'd have lots of simple, composable units and just a few compositional units - code coverage will probably be around 30%-ish? So I have a hard time coming up with a good value.

Maybe having this metric doesn't mean anything for us at the moment as long as it's not 0%.